assets/js/googlesitekit/modules/datastore/settings.test.js

```diff
@@ -0,0 +1,123 @@
+			store,
+			{
+				selectors: {
+					isModuleActive,
+				},
+			},
+			settings,
+		) );
+
+		registry.registerStore( moduleStoreName, Data.combineStores(
+			storeDefinition,
+		) );
+	} );
+
+	describe( 'reducer', () => {
+		it( 'has the appropriate initial state', () => {
+			expect( true ).toBeTruthy();
+			const initialState = {};
+			const initialStoreState = store.getState();
+
+			// Ensure we have default state correct
+			expect( Object.keys( initialStoreState ) ).toContain( 'panelState' );
+			expect( initialStoreState.panelState ).toEqual( initialState );
+		} );
+	} );
+
+	describe( 'selectors', () => {
+		describe( 'getModuleSettingsPanelState', () => {
+			it( 'gets module settings panel state', () => {
+				// Default state for valid module is closed
+				expect( registry.select( STORE_NAME ).getModuleSettingsPanelState( slug ) ).toBe( 'closed' );
+				// Default state for invalid module is closed
+				expect( registry.select( STORE_NAME ).getModuleSettingsPanelState( nonExistentModuleSlug ) ).toBe( null );
+
+				registry.dispatch( STORE_NAME ).setModuleSettingsPanelState( slug, 'view' );
+				expect( registry.select( STORE_NAME ).getModuleSettingsPanelState( slug ) ).toBe( 'view' );
+
+				registry.dispatch( STORE_NAME ).setModuleSettingsPanelState( slug, 'closed' );
+				expect( registry.select( STORE_NAME ).getModuleSettingsPanelState( slug ) ).toBe( 'closed' );
+
+				registry.dispatch( STORE_NAME ).setModuleSettingsPanelState( slug, 'edit' );
+				expect( registry.select( STORE_NAME ).getModuleSettingsPanelState( slug ) ).toBe( 'edit' );
+
+				// If the value is edit, all other module settings panels which currently have a status of view or edit should automatically be set to locked.
+				registry.dispatch( STORE_NAME ).setModuleSettingsPanelState( nonExistentModuleSlug, 'edit' );
+				expect( registry.select( STORE_NAME ).getModuleSettingsPanelState( nonExistentModuleSlug ) ).toBe( 'edit' );
+				expect( registry.select( STORE_NAME ).getModuleSettingsPanelState( slug ) ).toBe( 'locked' );
+
+				// If the value is closed or view and previously the value was edit, all other module settings panels which currently have a status of locked should automatically be set to view.
+				registry.dispatch( STORE_NAME ).setModuleSettingsPanelState( nonExistentModuleSlug, 'closed' );
+				expect( registry.select( STORE_NAME ).getModuleSettingsPanelState( nonExistentModuleSlug ) ).toBe( 'closed' );
+				expect( registry.select( STORE_NAME ).getModuleSettingsPanelState( slug ) ).toBe( 'view' );
+			} );
+		} );
+	} );
+
+	describe( 'actions', () => {
+		describe( 'setModuleSettingsPanelState', () => {
+			it( 'sets module settings panel state', () => {
+				registry.dispatch( STORE_NAME ).setModuleSettingsPanelState( slug, 'closed' );
+				let expectedPanelState = { ...store.getState().panelState, [ slug ]: 'closed' };
+				expect( store.getState().panelState ).toEqual( expectedPanelState );
```

Comment:
I think these can be simplified a bit.
```suggestion
				expect( store.getState().panelState[slug] ).toEqual( 'closed' );
```
Perhaps even an `it.each` would be useful to avoid duplication here.